src/utils/magitUtils.ts

```diff
@@ -95,6 +97,16 @@ export default class MagitUtils {
     return window.showQuickPick(refs, { placeHolder: prompt });
   }
 
+  public static async chooseCommit(repository: MagitRepository, prompt: string) {
+
+    const log =
+      repository.log({ maxEntries: 100 })
+        .then(log => log.map(commit => { return { label: commit.hash, description: commit.message }; }));
```

Comment:
Suggestion: presenting the short hash of the commit in the menu.
```suggestion
        .then(log => log.map(commit => { return { label: GitTextUtils.shortHash(commit.hash), description: commit.message }; }));
```

Reply:
Hmmmmmm...That would prevent someone from pasting in a full hash and finding the correct hash. Perhaps what I could do is put the full hash at the end of the description, and make the label the shorthash?

Reply:
Right, I see.
Another way could be to use the `QuickMenuUtil.showMenuWithFreeform`
Accepting pick-items, but if there are no matches, accept whatever is in the input field.

Right now this is used for branch checkout. If you do `b b` and paste in a commit/ref/shorthash/whatever, it will attempt to check out that.
Simlar to how helm+magit works. Not the best discoverability though.

Reply:
I ended up going that route, it is cosmetically better to just see the short hash, and the long hash is appended to the message so you can search it. Good call :)

Reply:
In retrospect, I think your most recent comment (we just crossed signals there) is a good point. This doesn't show every commit, only the last ones. So we need a way to allow fixing up a really old commit, which would be just accepting non-match as the input. I'll set that up.